lib/model-factory.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const { decorate, metadata, param } = require('./helpers.js');
+const repository = require('@loopback/repository');
+
+module.exports = function (modelName, modelDefinition) {
+  let classes = {};
+
+  classes[modelName] = class extends repository.Entity {
+    constructor(data) {
+      super(data);
+    }
+  };
+
+  let Model = classes[modelName];
+  Object.defineProperty(Model, 'name', { writable: true });
+  Model.name = modelName;
+  Object.defineProperty(Model, 'name', { writable: false });
+
+  // More to be added later
+  const typeMap = {
+    'string': String,
+    'number': Number
+  };
+
+  const modelProperties = modelDefinition.properties;
+  Object.keys(modelProperties).forEach(key => {
+    const property = modelProperties[key];
+    let repositoryProperties = {
+      type: property.type
+    };
+    if (key === 'id') {
+      repositoryProperties.id = true;
+    }
+    decorate([
+      repository.property(repositoryProperties),
+      metadata('design:type', typeMap[property.type])
```

Comment:
I would like LB4 to work without `design:type` metadata too. Where are we relying on `design:type` of model properties? I'd like us to rework those places to use the real model definition instead. IMO, the only place accessing design-type metadata of model properties is the `@property` and `@model` decorators that are building the actual model definition.

Reply:
`design:type` is only used for inference/inspection of TS types and they can/should be always overridable.